### DISCUSSION TITLE
Issue with Module Registry

### DISCUSSION THREAD

**Discussion by jderieg** [terrakube-io/terrakube#469] [Q&A] [answered]
I'm attempting to host a module in Terrakube that connects out to GitHub. I have the provider integration set up correctly, as it is able to connect out and get the release version and other data, but for some reason it is not loading any of the resource data (Readme, Inputs, Outputs, Resources). It just says 'loading...'
Any ideas of where to look to troubleshoot?

![tfmodule](https://github.com/AzBuilder/terrakube/assets/63372114/b4285b14-ea16-4cb0-bd9e-227fa81a490d)

**Comment by alfespa17** [comment]
Hello @jderieg 

That usually happens if you are using a repository that has no "tags/releases". Could you check if the repo has "tags/releases"?

**Reply by alfespa17** [reply]
You have tags in the repository, I can see from this part of the image. Terrakube was able to read that information and connect to the repository.
![image](https://github.com/AzBuilder/terrakube/assets/4461895/199a1b25-a485-471e-be2c-2278d5c6d456)

You could check the pod logs for the registry.

```bash
kubectl get pods -n your_terrakube_namespace
kubectl logs pod-name-registry -n your_terrakube_namespace
```

**Reply by jderieg** [reply]
Hmm, looks like maybe a string parser error? Are spaces allowed in the module name? 

```
2023-08-17 18:19:54.940 ERROR 1 --- [nio-8075-exec-8] o.a.c.c.C.[.[.[/].[dispatcherServlet]    : Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception [Request processing failed; nested exception is feign.FeignException$BadRequest: [400 ] during [GET] to [http://terrakube-api-service:8080/api/v1/organization/c0842cc6-3f3b-401f-8f49-e9a5859ca282/module?filter%5Bmodule%5D=name==Terraform%20Modules;provider==aws] [TerrakubeClient#getModuleByNameAndProvider(String,String,String)]: [{"errors":[{"detail":"Invalid query parameter: filter[module]\ncz.jirutka.rsql.parser.ParseException: Encountered &#34; &lt;UNRESERVED_STR&gt; &#34;Devops &#34;&#34; at line 1, column 11.\nWas expecting one of:\n    &lt;EOF&gt; \n    &lt;AND&gt; ...\n    &lt;OR&gt; ...\n    "}]}]] with root cause

feign.FeignException$BadRequest: [400 ] during [GET] to [http://terrakube-api-service:8080/api/v1/organization/c0842cc6-3f3b-401f-8f49-e9a5859ca282/module?filter%5Bmodule%5D=name==Terraform%20Modules;provider==aws] [TerrakubeClient#getModuleByNameAndProvider(String,String,String)]: [{"errors":[{"detail":"Invalid query parameter: filter[module]\ncz.jirutka.rsql.parser.ParseException: Encountered &#34; &lt;UNRESERVED_STR&gt; &#34;Devops &#34;&#34; at line 1, column 11.\nWas expecting one of:\n    &lt;EOF&gt; \n    &lt;AND&gt; ...\n    &lt;OR&gt; ...\n    "}]}]
```

**Reply by alfespa17** [reply, marked as answer]
That is correct don't use "spaces" in the module name, you could use "-" like this:

![image](https://github.com/AzBuilder/terrakube/assets/4461895/bdadec53-b460-42c1-a8ab-ab141d9e442d)

**Reply by jderieg** [reply]
Yup, that fixed it. Thanks! I'm still not seeing any resources under the resources tab. Not sure why. I also opened a feature request to list submodules: https://github.com/orgs/AzBuilder/discussions/470

**Reply by alfespa17** [reply]
I think you dont see any data under "resources" because we dont support submodules inside the registry ,right now there is a 1-to-1 relationship between a repository and module.

**Reply by jderieg** [reply]
ok, makes sense. Thanks for the explanation.
